src/main/java/wtf/meier/tariff/interpreter/model/tariff/calculator/SlotBasedTariffCalculator.kt

```diff
@@ -16,30 +17,34 @@ class SlotBasedTariffCalculator(
     private val rateCalculator: RateCalculator = RateCalculator()
 ) {
 
-    fun calculate(tariff: SlotBasedTariff, rentalStart: Instant, rentalEnd: Instant): Receipt {
+    fun calculate(tariff: SlotBasedTariff, rentalPeriod: RentalPeriod): Receipt {
         val positions = mutableListOf<RateCalculator.CalculatedPrice>()
+        positions.addAll(rentalPeriod.positions)
         val rateMap = tariff.rates.associateBy { it.id }
 
-        var slotStart = rentalStart
+        var slotStart: Instant = rentalPeriod.calculatedStart
+
```

Comment:
We don't need to be explict about the type here, do we?

Reply:
No we don't.